src/Common/tests/gtest_slru_cahce.cpp

```diff
@@ -0,0 +1,119 @@
+#include <iomanip>
+#include <iostream>
+#include <gtest/gtest.h>
+#include <Common/SLRUCache.h>
+
+TEST(SLRUCache, set)
```

Comment:
Let's add `SLRUCache` under setting for, for example, uncompressed cache (see UncompressedCache.h). See where it is used and add a setting like `uncompressed_cache_policy`, by default LRU. The test should check profile events (`UncompressedCacheHits`, `UncompressedCacheMisses`...), you can take them from `system.query_log` .(`select ProfileEvents['UncompressedCacheHits'] where query like '...'`) For test base example you can take [02124_uncompressed_cache.sql](https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/02124_uncompressed_cache.sql), but need to make it more complex, check different queries, settings...